client/src/main/java/com/defold/extender/client/ExtenderClient.java

```diff
@@ -63,25 +67,18 @@ public class ExtenderClient {
      * @param cacheDir        A directory where the cache files are located (it must exist beforehand)
      */
     public ExtenderClient(String extenderBaseUrl, File cacheDir) throws IOException {
-        this(new DefaultHttpClient(), extenderBaseUrl, cacheDir);
-    }
-
-    /**
-     * Creates an extender client
-     *
-     * @param httpClient      The http client instance to use when communicating with the extender server
-     * @param extenderBaseUrl The build server url (e.g. https://build.defold.com)
-     * @param cacheDir        A directory where the cache files are located (it must exist beforehand)
-     */
-    public ExtenderClient(AbstractHttpClient httpClient, String extenderBaseUrl, File cacheDir) throws IOException {
         this.extenderBaseUrl = extenderBaseUrl;
         this.cache = new ExtenderClientCache(cacheDir);
         this.httpCookies = new BasicCookieStore();
         this.buildSleepTimeout = Long.parseLong(System.getProperty("com.defold.extender.client.build-sleep-timeout", "5000"));
         this.buildResultWaitTimeout = Long.parseLong(System.getProperty("com.defold.extender.client.build-wait-timeout", "1200000"));
         this.headers = new ArrayList<BasicHeader>();
-        this.httpClient = httpClient;
-        this.httpClient.setCookieStore(httpCookies);
+        RequestConfig requestConfig = RequestConfig.custom()
+            .setExpectContinueEnabled(true).build();
```

Comment:
Second part of the fix. HTTP client waits 100 Continue from server side in case of additional authentication before sending request body. It helps avoid `Broken pipe` error, when client received 401 on `/build**` and tried push request body.